packages/aws-cdk-lib/aws-rds/README.md

```diff
@@ -1242,3 +1242,87 @@ new rds.DatabaseCluster(this, 'DatabaseCluster', {
   preferredMaintenanceWindow: 'Sat:22:15-Sat:22:45',
 });
 ```
+
+## Performance Insights
+
+You can enable Performance Insights for a clustered database or an instance database.
+
+### Clustered Database
+
+You can enable Performance Insights at cluster level or instance level.
+
+To enable Performance Insights at the cluster level, set the `enablePerformanceInsights` property for the `DatabaseCluster` to `true`.
+If you want to specify the detailed settings, you can use the `performanceInsightRetention` and `performanceInsightEncryptionKey` properties.
+
+The settings are then applied to all instances in the cluster.
+
+```ts
+declare const vpc: ec2.Vpc;
+declare const kmsKey: kms.Key;
+new rds.DatabaseCluster(this, 'Database', {
+  engine: rds.DatabaseClusterEngine.AURORA,
+  vpc: vpc,
+  enablePerformanceInsights: true,
+  performanceInsightRetention: rds.PerformanceInsightRetention.LONG_TERM,
+  performanceInsightEncryptionKey: kmsKey,
+  writer: rds.ClusterInstance.provisioned('Writer', {
+    instanceType: ec2.InstanceType.of(ec2.InstanceClass.R7G, ec2.InstanceSize.LARGE),
+  }),
+});
+```
+
+To enable Performance Insights at the instance level, set the same properties for each instance of the `writer` and the `readers`.
+
+In this way, different settings can be applied to different instances in a cluster.
+
+However, if Performance Insights is enabled at the cluster level, it is not possible to specify a different value for the instance
+level than the cluster level.
```

Comment:
If Performance Insights was enabled at cluster and instance level, and the values of the `performanceInsightRetention` or `performanceInsightEncryptionKey` were different for the cluster and the instance, CFn failed.

If `performanceInsightRetention`s are different, CFn occurred:

```
Resource handler returned message: "PerformanceInsightsRetentionPeriod conflicts with cluster level parameter. ...“
```

If `performanceInsightEncryptionKey`s are different, CFn occurred:

```
Resource handler returned message: "PerformanceInsightsKMSKeyId conflicts with cluster level parameter. …”
```

However, if Performance Insights was enabled at cluster and instance level and the values were **the same** for the cluster and the instance, **CFn did not fail.**

If the value of `enablePerformanceInsights` was different for the cluster and the instance, it **did not result in an error**. If it was true in the cluster and **false in the instance**, it did not result in an error but was also **enabled in the instance**.

===

I wanted to implement validation of this behaviour. However, the instance does not have Performance Insights settings publicly, so it is not accessible from the cluster. Therefore, to implement validation, the structure of the instances had to be changed significantly, so I decided not to do it in this PR.

Reply:
> the structure of the instances had to be changed significantly,

Isn't exposing the props in [`AuroraClusterInstance`](https://github.com/aws/aws-cdk/blob/1726abddbe10a22c092de7bf0b566710d13f5efd/packages/aws-cdk-lib/aws-rds/lib/aurora-cluster-instance.ts#L444) and adding validation in [`validateClusterInstances`](https://github.com/aws/aws-cdk/blob/1726abddbe10a22c092de7bf0b566710d13f5efd/packages/aws-cdk-lib/aws-rds/lib/cluster.ts#L795) enough? 🤔

Reply:
I added the validation, what do you think? 

The cluster has a writer and readers as `IAuroraClusterInstance` not `AuroraClusterInstance`, so I added the variable `performanceInsightsEnabled` in `IAuroraClusterInstance` interface. 

And the `legacyCreateInstances` function for the database cluster is not a method, I also added a public variable `performanceInsightsEnabled` in `DatabaseClusterNew`.

https://github.com/aws/aws-cdk/pull/31385/commits/cf568902909eb3e0ee5ec2abed91623d08bc7363